activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -90,16 +90,20 @@ def expand_from_hash(attributes)
             queries.reduce(&:or)
           elsif table.aggregated_with?(key)
             mapping = table.reflect_on_aggregation(key).mapping
-            queries = Array.wrap(value).map do |object|
-              mapping.map do |field_attr, aggregate_attr|
-                if mapping.size == 1 && !object.respond_to?(aggregate_attr)
-                  build(table.arel_attribute(field_attr), object)
-                else
+            if mapping.length == 1
+              column_name, aggr_attr = mapping.first
```

Comment:
I'd prefer using the same variable names in both path.
```suggestion
              field_attr, aggregate_attr = mapping.first
```

Reply:
That causes "shadowing outer local variable" ruby warnings

Reply:
oh... I see...